handler/src/main/java/io/netty/handler/pcap/PcapWriter.java

```diff
@@ -58,22 +62,25 @@
      * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
      */
     void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
-        long currentTime = System.currentTimeMillis();
+        long timestamp = System.currentTimeMillis();
 
         PcapHeaders.writePacketHeader(
                 packetHeaderBuf,
-                (int) TimeUnit.MILLISECONDS.toSeconds(currentTime),
-                (int) TimeUnit.MILLISECONDS.toMicros(currentTime) % 1000000,
+                (int) (timestamp / 1000L),
+                (int) (timestamp % 1000L * 1000L),
                 packet.readableBytes(),
                 packet.readableBytes()
         );
 
-        packetHeaderBuf.readBytes(outputStream, packetHeaderBuf.readableBytes());
-        packet.readBytes(outputStream, packet.readableBytes());
+        if (!isClosed) {
```

Comment:
Would be better to hoist this condition to the start of the method and return early if the writer is closed, no?